src/psyclone/tests/domain/common/extract_driver_creator_test.py

```diff
@@ -0,0 +1,547 @@
+    with pytest.raises(TransformationError) as err:
+        edc.add_all_kernel_symbols(schedule_copy, symbol_table)
+    assert "Error when constructing driver for 'invoke_0_compute_kernel': " \
+           "Unknown intrinsic data type 'Intrinsic.INTEGER'" in str(err.value)
+
+
+# -----------------------------------------------------------------------------
+def test_driver_creation_same_symbol(tmpdir):
+    '''Make sure that if a symbol appears in more than one invoke no duplicated
+    symbol is created.
+
+    '''
+    # Use tmpdir so that the driver is created in tmp
+    tmpdir.chdir()
+
+    etrans = GOceanExtractTrans()
+    # The fourth invoke calls a kernel twice with identical parameters
+    psy, invoke = get_invoke("driver_test.f90",
+                             GOCEAN_API, idx=3, dist_mem=False)
+    schedule = invoke.schedule
+
+    etrans.apply(schedule.children[0], {'create_driver': True})
+    # We are only interested in the driver, so ignore results.
+    str(psy.gen)
+
+    driver = tmpdir.join("driver-psy_extract_example_with_various_variable_"
+                         "access_patterns-invoke_3:compute_kernel_code:r0.f90")
+    assert driver.isfile()
+
+    with driver.open("r") as driver_file:
+        driver_code = driver_file.read()
+    # Make sure we have both kernel calls in the driver.
+    correct = """  do j = out_fld_internal_ystart, out_fld_internal_ystop, 1
+    do i = out_fld_internal_xstart, out_fld_internal_xstop, 1
+      call compute_kernel_code(i, j, out_fld, in_out_fld, in_fld, dx, """ \
+      """in_fld_grid_dx, in_fld_grid_gphiu)
+    enddo
+  enddo
+  do j = out_fld_internal_ystart, out_fld_internal_ystop, 1
+    do i = out_fld_internal_xstart, out_fld_internal_xstop, 1
+      call compute_kernel_code(i, j, out_fld, in_out_fld, in_fld, dx, """ \
+      """in_fld_grid_dx, in_fld_grid_gphiu)
+    enddo
+  enddo"""
+    assert correct in driver_code
+
+
+# -----------------------------------------------------------------------------
+def test_driver_creation_import_modules(fortran_reader):
+    '''Tests if global and local interfaces in calls are handled correctly.
+    '''
+
+    code = '''program test_prog
+              use my_module, only : mod_func
+              call mod_func()
+              call ext_func()
+              call mod_func()
+              end program test_prog'''
+    psyir = fortran_reader.psyir_from_source(code)
+    program = psyir.children[0]   # psyir is a FileContainer, take the program
+    edc = ExtractDriverCreator()
+    # Delete all symbols in the symbol table so we can check if the right
+    # symbols are added:
+    program.scope._symbol_table = SymbolTable()
+    edc.import_modules(program, psyir.children[0])
+    symbol_table = program.scope.symbol_table
+    all_symbols = symbol_table.get_symbols()
+    assert len(all_symbols) == 2
+    assert str(all_symbols["my_module"]) == "my_module: <not linked>"
+    mod_func = all_symbols["mod_func"]
```

Comment:
``ext_func`` is not declared in the original program, so nothing we can do about that.

Reply:
OK. I'm surprised that the frontend doesn't reject this as there's no way it can be brought into scope. I'll ask on Teams.

Reply:
Turns out it was me who modified the frontend to accept such a thing!

Reply:
I've tried to update this code - adding ``implicit none`` does not trigger a warning/error from the parser (I thought it would). But if I add an ``external :: ext_func``, then the parsing breaks down, I get a code block for everything :(